packages/origin/contracts/Origin/CertificateLogic.sol

```diff
@@ -158,6 +158,39 @@ contract CertificateLogic is CertificateInterface, CertificateSpecificContract,
         }
     }
 
+    /// @notice claims a set of certificates
+    /// @param _idArray the ids of the certificates to be claimed
+    function claimCertificateBulk(uint[] calldata _idArray) external {
+        for (uint i = 0; i < _idArray.length; i++) {
```

Comment:
This potentially can run out of gas, UI seems to limit to 100 items per call. I would check the upper limit on this.